extensions/cornerstone/src/services/ViewportService/CornerstoneViewportService.ts

```diff
@@ -320,29 +362,39 @@ class CornerstoneViewportService implements IViewportService {
 
     this.viewportsDisplaySets.set(viewport.id, [displaySetInstanceUID]);
 
-    let initialImageIndexToUse = initialImageIndex;
+    let initialImageIndexToUse =
+      presentation?.initialImageIndex ?? initialImageIndex;
 
     if (!initialImageIndexToUse) {
       initialImageIndexToUse =
         this._getInitialImageIndexForStackViewport(viewportInfo, imageIds) || 0;
     }
 
-    const { voi, voiInverted } = displaySetOptions[0];
-    const properties = {};
-    if (voi && (voi.windowWidth || voi.windowCenter)) {
-      const { lower, upper } = csUtils.windowLevel.toLowHighRange(
-        voi.windowWidth,
-        voi.windowCenter
-      );
-      properties.voiRange = { lower, upper };
-    }
+    const properties = presentation?.properties || {};
+    if (!presentation?.properties) {
+      const { voi, voiInverted } = displaySetOptions[0];
+      if (voi && (voi.windowWidth || voi.windowCenter)) {
+        const { lower, upper } = csUtils.windowLevel.toLowHighRange(
+          voi.windowWidth,
+          voi.windowCenter
+        );
+        properties.voiRange = { lower, upper };
+      }
 
-    if (voiInverted !== undefined) {
-      properties.invert = voiInverted;
+      if (voiInverted !== undefined) {
+        properties.invert = voiInverted;
+      }
     }
 
-    viewport.setStack(imageIds, initialImageIndexToUse).then(() => {
+    // There is a bug in CS3D that the setStack does not
+    // navigate to the desired image.
+    viewport.setStack(imageIds, 0).then(() => {
+      // The scroll, however, works fine in CS3D
+      viewport.scroll(initialImageIndexToUse);
       viewport.setProperties(properties);
+      const { zoom, pan } = presentation || {};
+      if (zoom) viewport.setZoom(zoom);
```

Comment:
I wonder if simply getting and setting the camera is best for a stack viewport because I think all of pan, zoom, flip and rotate are captured there. This is something we can leave as a TODO.

Reply:
Done.